src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,165 @@
+        fieldName: string;
+        accessorName: string;
+        accessorType: TypeNode;
+        propertyDeclaration: PropertyDeclaration;
+        needUpdateName: boolean;
+        hasModifiers: boolean;
+        needUpdateModifiers: boolean;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        return [
+            {
+                name: actionName,
+                description: actionDescription,
+                actions: [
+                    {
+                        name: actionName,
+                        description: actionDescription
+                    }
+                ]
+            }
+        ];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+        const { fieldName, accessorName, accessorType, propertyDeclaration, needUpdateName, hasModifiers, needUpdateModifiers } = fieldInfo;
+        const accessorModifiers = hasModifiers ? (
+            (getModifierFlags(propertyDeclaration) & ModifierFlags.Private || !propertyDeclaration.modifiers) ? createNodeArray([createToken(SyntaxKind.PublicKeyword)]) : propertyDeclaration.modifiers
+        ) : undefined;
+
+        const getAccessor = generateGetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+        const setAccessor = generateSetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+
+        const modifiers = hasModifiers ? createNodeArray([createToken(SyntaxKind.PrivateKeyword)]) : undefined;
+        if (needUpdateName || needUpdateModifiers) {
+            changeTracker.replaceNode(file, propertyDeclaration, updateoriginalPropertyDeclaration(propertyDeclaration, fieldName, modifiers), {
+                suffix: newLineCharacter
+            });
+        }
+
+        changeTracker.insertNodeAfter(file, propertyDeclaration, getAccessor);
+        changeTracker.insertNodeAfter(file, propertyDeclaration, setAccessor);
+
+        return {
+            edits: changeTracker.getChanges(),
+            renameFilename: undefined,
+            renameLocation: undefined,
+        };
+    }
+
+    function getConvertibleFieldAtPosition(file: SourceFile, startPosition: number): Info | undefined {
+        const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
+        const propertyDeclaration = findAncestor(node.parent, isPropertyDeclaration);

```

Comment:
we also need to handle parameter properties. so this can be a property or a parameter declaration.